llarp/handlers/tun.cpp

```diff
@@ -407,7 +407,24 @@ namespace llarp
         service::Address addr;
         if (addr.FromString(qname, ".loki") || addr.FromString(qname, ".snode")
             || is_random_snode(msg) || is_localhost_loki(msg))
+        {
           msg.AddMXReply(qname, 1);
+        }
+        else if (service::NameIsValid(lnsName))
+        {
+          return LookupNameAsync(
+              lnsName, [msg = std::make_shared<dns::Message>(msg), lnsName, reply](auto maybe) {
```

Comment:
I don't think this needs to be a shared_ptr, just capture `msg` (to copy it) should be enough?

Reply:
you're right. fixed, needed a mutable lambda